include/openenclave/internal/rdrand.h

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+#ifndef _OE_RDRAND_H
+#define _OE_RDRAND_H
+
+#include <openenclave/bits/types.h>
+
+OE_EXTERNC_BEGIN
+
+/**
+ * Generate a sequence of random bytes using Intel RDRAND instruction
+ *
+ * This function generates 8 random bytes using direct call to Intel's RDRAND
+ * instruction. This method will block if there is insufficient hardware
+ * entropy to provide the full 64-bits of randomness.
+ *
+ * @return OE_OK on success
+ */
+uint64_t oe_rdrand(void);
```

Comment:
For my understanding, why does this need to be in a new header instead of random.h?

Reply:
This isn't part of the crypto API. It's specific to oecore and x86_64 for generating a entropy seed for the platform.